backend/src/v5/models/metadata.js

```diff
@@ -33,12 +32,12 @@ Metadata.getMetadataById = async (teamspace, model, metadataId, projection) => {
 	return metadata;
 };
 
-Metadata.updateCustomMetadata = async (teamspace, model, metadataId, dataToUpdate) => {
-	const { metadata } = await Metadata.getMetadataById(teamspace, model, metadataId, { metadata: 1 });
-	const metadataKeyLookup = new UUIDLookUpTable(metadata.map(({key}) => key));
+Metadata.updateCustomMetadata = async (teamspace, model, metadataId, changeSet) => {
+	const { metadata } = await Metadata.getMetadataByQuery(teamspace, model, { _id: metadataId }, { metadata: 1 });
+	const metadataKeyLookup = metadata.reduce((a, b) => ({ ...a, [b.key]: 1 }), {});
 
-	dataToUpdate.forEach((um) => {
-		const metadataExists = metadataKeyLookup.has(um.key);
+	changeSet.forEach((um) => {
+		const metadataExists = metadataKeyLookup[um.key];
 		if (metadataExists) {
 			const existingMetadata = metadata.find((m) => m.key === um.key);
```

Comment:
whole point of using a lookup table is to get rid of this line?

if you need to know the index then you should just set the index on the reduce call instead of setting it to 1.